### ISSUE TITLE
InputNumber: JavaScript errors when used inside composite component without setting attributes

### ISSUE THREAD

**Issue by mkomko** [primefaces/primefaces#6019]
**Describe the defect**
After upgrading to PrimeFaces 7.0.14 from 7.0.10 our inputNumber composite component does not work anymore. Whenever one of the attributes decimalPlaces, decimalSeparator, thousandSeparator, minValue or maxValue is not set when defining the composite component, and thus the attributes are empty when passed to the PrimeFaces inputNumber component, JavaScript errors occur:

```
Uncaught TypeError: Cannot read property 'replace' of undefined
    at bK.bB (inputnumber.js.xhtml?ln=primefaces&v=8.0:14)
```

```
Uncaught SyntaxError: Invalid regular expression: /[^-0123456789\]/: Unterminated character class
    at new RegExp (<anonymous>)
    at Function.Y (inputnumber.js.xhtml?ln=primefaces&v=8.0:14)
```

**Reproducer**
You can find the reproducer here using primefaces-test.git:
[primefaces-test-input-number.zip](https://github.com/primefaces/primefaces/files/4757427/primefaces-test-input-number.zip)

**Environment:**
 - PF Version: _7.0.14_
 - JSF + version: Mojarra 2.2.20
 - Affected browsers: _ALL_

**To Reproduce**
1. Start the reproducer.
2. Check the JavaScript console. One error is from the first component, one from the second.

**Expected behavior**
I expected our composite component to function as before the upgrade.

**Example XHTML**
```html
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:ui="http://java.sun.com/jsf/facelets"
      xmlns:f="http://java.sun.com/jsf/core"
      xmlns:p="http://primefaces.org/ui"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:comp="http://xmlns.jcp.org/jsf/composite/components/test">

<h:head>
    <title>PrimeFaces Test</title>
</h:head>
<h:body>

    <h1>#{testView.testString}</h1>
    <h:form id="frmTest">
        <comp:inputNumber id="test"
                          value="#{testView.testValue}"/>
        <comp:inputNumber id="test2"
                          value="#{testView.testValue}"
                          decimalPlaces="0"
                          minValue="0"/>
    </h:form>

</h:body>
</html>
```

```html
<?xml version="1.0" encoding="UTF-8"?>

<ui:component xmlns="http://www.w3.org/1999/xhtml"
              xmlns:h="http://xmlns.jcp.org/jsf/html"
              xmlns:cc="http://xmlns.jcp.org/jsf/composite"
              xmlns:ui="http://xmlns.jcp.org/jsf/facelets"
              xmlns:p="http://primefaces.org/ui"
              xmlns:comp="http://xmlns.jcp.org/jsf/composite/components/test">

    <cc:interface>
        <cc:attribute name="value"/>

        <cc:attribute name="decimalPlaces"/>
        <cc:attribute name="decimalSeparator"/>
        <cc:attribute name="thousandSeparator"/>
        <cc:attribute name="minValue"/>
        <cc:attribute name="maxValue"/>

    </cc:interface>

    <cc:implementation>
        <div id="#{cc.clientId}">

            <p:inputNumber id="input"
                           value="#{cc.attrs.value}"
                           inputStyle="#{cc.attrs.inputStyle}"
                           decimalPlaces="#{cc.attrs.decimalPlaces}"
                           decimalSeparator="#{cc.attrs.decimalSeparator}"
                           thousandSeparator="#{cc.attrs.thousandSeparator}"
                           minValue="#{cc.attrs.minValue}"
                           maxValue="#{cc.attrs.maxValue}"/>

        </div>
    </cc:implementation>
</ui:component>
```

**Example Bean**
```java
package org.primefaces.test;

import java.io.Serializable;
import javax.annotation.PostConstruct;
import javax.faces.view.ViewScoped;
import javax.inject.Named;

@Named
@ViewScoped
public class TestView implements Serializable {
    
    private String testString;

    private Long testValue;
    private Long testValue2;
    
    @PostConstruct  
    public void init() {
        testString = "Welcome to PrimeFaces!!!";
    }

    public String getTestString() {
        return testString;
    }

    public void setTestString(String testString) {
        this.testString = testString;
    }

    public Long getTestValue() {
        return testValue;
    }

    public void setTestValue(Long testValue) {
        this.testValue = testValue;
    }

    public Long getTestValue2() {
        return testValue2;
    }

    public void setTestValue2(Long testValue2) {
        this.testValue2 = testValue2;
    }
}

```

It would be nice to get it to work as it did before. But if you have any suggestions as to how to modify the composite component to achieve a working state I'll take them.
I maybe could set default values for the attributes or only set the attributes on the inputNumber component when they are not empty by using c:if and f:attribute (dirty hack).

Thank you very much in advance!

**Comment by melloware** [comment]
The issue is we upgraded to the latest AutoNumeric plugin and some values you are providing can't be empty string "" anymore.  So change your composite component to this and it fixes it.
```xml
    <cc:interface>
        <cc:attribute name="value"/>
        <cc:attribute name="decimalPlaces" default="2" />
        <cc:attribute name="decimalSeparator" default="."/>
        <cc:attribute name="thousandSeparator" default=","/>
        <cc:attribute name="minValue" default="-9999999999"/>
        <cc:attribute name="maxValue" default="9999999999"/>
    </cc:interface>
```

I will also be submitting a PR to fix if decimalSeaprator is "" is causes this Regexp JS error.

**Comment by mkomko** [comment]
Thank you, I can do that. Wouldn't the correct minValue and maxValue be -10000000000000 and 10000000000000 respectively, though?

**Comment by melloware** [comment]
you can make it whatever you want I was just throwing numbers in there to get the JS errors to go away.

**Comment by mkomko** [comment]
Haha, okay. I would of course want them to be what the internal inputNumber defaults are. Those seem to be it. Thanks!

**Comment by mkomko** [comment]
@melloware Would it be possible to get your fix for the decimalSeparator in 7.0.15?
Thank you very much in advance!

**Comment by melloware** [comment]
I don't work for PrimeTek so that's up to @mertsincan 

Also I reported this to AutoNumeric: https://github.com/autoNumeric/autoNumeric/issues/687
